src/db/spanner/pool.rs

```diff
@@ -55,9 +55,15 @@ impl SpannerDbPool {
 
     pub fn new_without_migrations(settings: &Settings, metrics: &Metrics) -> Result<Self> {
         let manager = SpannerConnectionManager::new(settings)?;
+        let max_size = settings.database_pool_max_size.unwrap_or(10);
+        // r2d2 creates max_size count of db connections on creation via its
+        // own thread_pool. increase its default size to quicken their
+        // creation, accommodating large max_size values (otherwise it may
+        // timeout)
+        let r2d2_thread_pool = ScheduledThreadPool::new((max_size as f32 * 0.05) as usize);
```

Comment:
I'm confused here. In the minimum case, we're taking 5% of 10, which rounds out to be 0. If we're trying to increase the available thread pools, why are we taking such a small percentage of the requested number?

Reply:
thanks, was going to add a max here and forgot